test/shopify-cli/theme/ignore_filter_test.rb

```diff
@@ -67,6 +67,24 @@ def test_patterns_to_regexes_and_globs
           assert_includes(filter.regexes, testcase[:regex]) unless testcase[:regex].nil?
         end
       end
+
+      def test_add
```

Comment:
Nitpick: I think we can write this one a bit more concisely:

```rb
def test_add
  filter = IgnoreFilter.new("/tmp")
  test_cases = [
    { pattern: "config/settings_data.json", glob: "*config/settings_data.json" },
    { pattern: "config/", glob: "*config/*" },
    { pattern: "*.jpg", glob: "*.jpg" },
    { pattern: "/\\.(txt|gif|bat)$/", regex: /\.(txt|gif|bat)$/ },
  ]

  test_cases.each do |test_case|
    filter.add_patterns([test_case.fetch(:pattern)])
    assert_includes(filter.globs, test_case.fetch(:glob)) unless test_case.key?(:glob)
    assert_includes(filter.regexes, test_case.fetch(:regex)) unless test_case.key?(:regex)
  end
end
```

Note that I'm using `.fetch(:key)` instead of `[:key]` to raise an error in case the key doesn't exist and prevent `nil` value from being passed and causing harder-to-debug errors down the road. You can also turn `test_case[:key].nil?` into `test_case.key?(:key)` to know whether a key is present in a hash.

Reply:
Fixed in 8e14fdf